esp/services/ws_dfu/ws_dfuXRefService.cpp

```diff
@@ -599,17 +611,19 @@ bool CWsDfuXRefEx::onDFUXRefList(IEspContext &context, IEspDFUXRefListRequest &r
         ForEachItemIn(c, clusters)
         {
             IConstWUClusterInfo &cluster = clusters.item(c);
+            bool isThorCluster = false;
             const StringArray &primaryThorProcesses = cluster.getPrimaryThorProcesses();
             ForEachItemIn(i,primaryThorProcesses)
             {
-                const char *thorProcess = primaryThorProcesses.item(i);
-                if (!thorProcess || !*thorProcess)
-                    continue;
-                bool* found = uniqueProcesses.getValue(thorProcess);
-                if (found && *found)
-                    continue;
-                uniqueProcesses.setValue(thorProcess, true);
-                addXRefNode(thorProcess, pXRefNodeTree);
+                if (addUniqueXRefNode(primaryThorProcesses.item(i), uniqueProcesses, pXRefNodeTree))
+                    isThorCluster = true;
```

Comment:
I find setting this flag and checking it a little less clear than just wrapping this section in:

`switch (cluster.getPlatform())`
`{`
`case ThorLCRCluster:`
`case RoxieCluster:`
`}`